app/controllers/application_controller.rb

```diff
@@ -359,6 +360,25 @@ def set_mobile_view
     session[:mobile_view] = params[:mobile_view] if params.has_key?(:mobile_view)
   end
 
+  def handle_safari_svc_session
+    # Workaround for dropped cookies in DiscourseHub SafariViewController iOS 12+.
```

Comment:
do we want to call this "safari" specific I agree safari is the symptom but really any client could opt for the same magic no?

Reply:
You're right, of course, no need to call this Safari anything. My initial hunch was to explain the use case for this now, but it's already usable by any client. (Which is also why there is an `auth_redirect` parameter, to let clients use their own URL here.)